### ISSUE TITLE
azure graph role tests cause NodeJs to OOM

### ISSUE THREAD

**Issue by balajikris** [Azure/azure-xplat-cli#3593]
See these failures:
1. https://travis-ci.org/Azure/azure-xplat-cli/builds/233814121?utm_source=github_status&utm_medium=notification
1. https://travis-ci.org/Azure/azure-xplat-cli/builds/233775082?utm_source=github_status&utm_medium=notification
1. last few failed builds from https://github.com/Azure/azure-xplat-cli/pull/3585

PR #3585 introduced this and I'm temporarily disabling these tests in #3592, Please re-enable them and send a PR which makes CI pass.

~~~
<--- Last few GCs --->
  615501 ms: Mark-sweep 1116.0 (1428.3) -> 1104.1 (1432.3) MB, 913.7 / 0.0 ms [allocation failure] [GC in old space requested].
  616633 ms: Mark-sweep 1104.1 (1432.3) -> 1104.0 (1434.3) MB, 1131.6 / 0.0 ms [allocation failure] [GC in old space requested].
  618040 ms: Mark-sweep 1104.0 (1434.3) -> 1120.6 (1403.3) MB, 1407.2 / 0.0 ms [last resort gc].
  619365 ms: Mark-sweep 1120.6 (1403.3) -> 1137.6 (1403.3) MB, 1324.7 / 0.0 ms [last resort gc].
<--- JS stacktrace --->
==== JS stack trace =========================================
Security context: 0x1ca360acfb51 <JS Object>
    1: toString [buffer.js:~487] [pc=0xc586ad890b1] (this=0xf44a9c12d09 <an Uint8Array with map 0x6fc5af06829>)
    5: parse [native json.js:~61] [pc=0xc586aa14381] (this=0x1ca360ac9111 <a JSON with map 0x2b86e56d82d1>,C=0xf44a9c12d09 <an Uint8Array with map 0x6fc5af06829>,w=0x1ca360a04381 <undefined>)
    6: arguments adaptor frame: 1->2
    7: initFromCmdMetadata [/home/travis/build/Azure/azu...
FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 2: 0x10983ec [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 3: v8::Utils::ReportApiFailure(char const*, char const*) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 5: v8::internal::Factory::NewRawTwoByteString(int, v8::internal::PretenureFlag) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 6: v8::internal::Factory::NewStringFromUtf8(v8::internal::Vector<char const>, v8::internal::PretenureFlag) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 7: v8::String::NewFromUtf8(v8::Isolate*, char const*, v8::String::NewStringType, int) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 8: node::StringBytes::Encode(v8::Isolate*, char const*, unsigned long, node::encoding) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
 9: void node::Buffer::StringSlice<(node::encoding)1>(v8::FunctionCallbackInfo<v8::Value> const&) [/home/travis/.nvm/versions/node/v6.10.0/bin/node]
10: 0xc586aa359f4
End:10:14:46 PM
Aborted (core dumped)
npm ERR! Test failed.  See above for more details.
The command "npm test" exited with 1.
~~~

**Comment by balajikris** [comment]
hey @darshanhs90 - I think I've addressed OOMs in our tests with this PR: https://github.com/Azure/azure-xplat-cli/pull/3600 .  Try getting latest changes from `dev` and see if your open PR to re-enable tests passes.

**Comment by balajikris** [comment]
Fixed with #3600 and tests are re-enabled with https://github.com/Azure/azure-xplat-cli/pull/3596
